drizzlepac/devutils/analyze_mvm_gaia_alignment.py

```diff
@@ -35,10 +34,102 @@
+    -------
+    detection_table : Astropy table
+        Table of sources detected by decutils.UserStarFinder()
+
+    matches_gaia_to_det : list
+        A list of the indices of GAIA sources that match detected sources
+
+    matches_det_to_gaia : list
+        A list of the indices of detected sources that match GAIA sources
+    """
+    log.setLevel(log_level)
+    daofind = decutils.UserStarFinder(fwhm=fwhm, threshold=0.0, coords=xy_gaia_coords,
+                                      sharplo=0.4, sharphi=0.9)
+    detection_table = daofind(mosaic_hdu["SCI"].data, mask=dao_mask_array)
+    detection_table.rename_column('xcentroid', 'X')
+    detection_table.rename_column('ycentroid', 'Y')
+    n_detection = len(detection_table)
+    n_gaia = len(gaia_table)
+    pct_detection = 100.0 * (float(n_detection) / float(n_gaia))
+    log.info("Found {} peaks from {} GAIA source(s)".format(n_detection, n_gaia))
+    log.info("{}% of GAIA sources detected".format(pct_detection))
+
+    # 4: convert UserStarFinder output x, y centroid positions to RA, DEC using step 1 WCS info
+    ra, dec = mosaic_wcs.all_pix2world(detection_table['X'], detection_table['Y'], 0)
+    ra_col = Column(name="RA", data=ra, dtype=np.float64)
+    dec_col = Column(name="DEC", data=dec, dtype=np.float64)
+    detection_table.add_columns([ra_col, dec_col], indexes=[3, 3])
+    if diagnostic_mode:
+        write_region_file("test_detection.reg", detection_table, ['RA', 'DEC'], log_level=log_level)
+
+    # 5: Identify and isolate X, Y, RA and DEC values common to both the gaia and detection tables.
+    # 5a: find sources common to both the gaia table and the detection table
+    try:
+        coo_prefix_string = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(6))
+        gaia_coo_filename = "{}_gaia.coo".format(coo_prefix_string)
+        det_coo_filename = "{}_det.coo".format(coo_prefix_string)
+        write_region_file(gaia_coo_filename, gaia_table, ['X', 'Y'], verbose=False)
+        write_region_file(det_coo_filename, detection_table, ['X', 'Y'], verbose=False)
+        matches_gaia_to_det, matches_det_to_gaia = cu.getMatchedLists([gaia_coo_filename, det_coo_filename],
+                                                                      [mosaic_imgname, mosaic_imgname],
+                                                                      [n_gaia, n_detection],
+                                                                      log_level)
+    finally:
+        for item in [det_coo_filename, gaia_coo_filename]:
+            if os.path.exists(item):
+                log.debug("Removing temp coord file {}".format(item))
+                os.remove(item)
+    return detection_table, matches_gaia_to_det, matches_det_to_gaia
+# ============================================================================================================
+
+
+def perform(mosaic_imgname, flcflt_list, diagnostic_mode=False, log_level=logutil.logging.INFO,
```

Comment:
The specification of 'flcflt_list' as a parameter is currently misleading.  This parameter has been implemented as a filename (str) whereas it has 'list' in the name and the docstring specifies it should be input as a list.  This parameter probably should be renamed so that both a list and a filename should (and can easily) be supported.  The docstring should be updated to correctly reflect the final implementation.

Reply:
issue resolved by commit [79addb4](https://github.com/spacetelescope/drizzlepac/pull/1241/commits/79addb45729f674dffa6c553bbb13dc70e46044b)